util/coverage/coverage_off_target.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""Measure unit test coverage of silicon creator code.
+
+Given a directory `out_root_dir` the coverage artifacts will be saved in
+`out_root_dir`/<HEAD_TIMESTAMP>-<HEAD_HASH>/unittest/`.
+
+Typical usage:
+    util/coverage/coverage_off_target.py --print-text-report
+"""
+
+import logging as log
+import subprocess
+from enum import Enum
+from pathlib import Path
+from random import randint
+from pprint import pformat
+from typing import List
+
+import typer
+
+app = typer.Typer(pretty_exceptions_enable=False, add_completion=False)
+
+# Commands that this script uses
+BAZEL = "./bazelisk.sh"
+LLD = "ld.lld"
+LLVM_COV = "llvm-cov"
+LLVM_PROFDATA = "llvm-profdata"
+
+# Title of the HTML coverage report
+REPORT_TITLE = "OpenTitan Unit Test Coverage"
+
+
+class LogLevel(str, Enum):
+    none = "none"
+    info = "info"
+    debug = "debug"
+
+
+def run(*args):
+    """Run the given command in a subprocess.
+
+    Args:
+        args: Command (the first parameter) and arguments (remaining arguments).
+
+    Returns:
+        Stdout lines in a list. Empty lines are filtered out.
+    """
+    log.debug(f"command: {' '.join(args)}")
+    try:
+        res = subprocess.run(args,
+                             stdout=subprocess.PIPE,
+                             stderr=subprocess.PIPE,
+                             encoding="utf-8",
+                             check=True)
```

Comment:
Nit: You could eliminate the exception handling by omitting `check=True` and directly checking the returned object's `returncode`.

Reply:
The resulting code would be similar though, wouldn't it?  I'd like to raise the error while printing the error.